File: lab_sheet_7/checkpoint_1/is_prime.py

import math

def is_prime(num):
    # eliminate even numbers early
    if num % 2 == 0:
        return False

    maxFactor = math.floor(math.sqrt(num))

    for i in range(2, maxFactor):
        if num % i == 0:
            return False

    return True

def main():
    num = int(input("Enter a number to test for primality: "))

    if is_prime(num):
        print("The number is prime")
    else:
        print("The number is not prime")

if __name__ == "__main__":
    main()

File: lab_sheet_8/checkpoint_8/IsPrime.java
import java.util.Scanner;

public class IsPrime {
    public static void main(String args[]) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter a number to test for primality: ");
        // int num = scanner.nextInt();
	int num = 72;

        if (isPrime(num))
            System.out.println("The number is prime.");
        else
            System.out.println("The number is not prime.");
    }

    public static boolean isPrime(int num) {
        // eliminate even numbers early
        if (num % 2 == 0)
            return false;

        int maxFactor = (int) Math.floor(Math.sqrt(num));

        for (int i = 2; i < maxFactor; i++) {
            if (num % i == 0)
                return false;
        }

        return true;
    }
}
